templates/repo/pulse.tmpl

```diff
@@ -0,0 +1,227 @@
+<h2 class="ui header activity-header">
+	<span>{{DateTime "long" .DateFrom}} - {{DateTime "long" .DateUntil}}</span>
+	<!-- Period -->
+	<div class="ui floating dropdown jump filter">
+		<div class="ui basic compact button">
+			{{ctx.Locale.Tr "repo.activity.period.filter_label"}} <strong>{{.PeriodText}}</strong>
+			{{svg "octicon-triangle-down" 14 "dropdown icon"}}
+		</div>
+		<div class="menu">
+			<a class="{{if eq .Period "daily"}}active {{end}}item" href="{{$.RepoLink}}/activity/daily">{{ctx.Locale.Tr "repo.activity.period.daily"}}</a>
+			<a class="{{if eq .Period "halfweekly"}}active {{end}}item" href="{{$.RepoLink}}/activity/halfweekly">{{ctx.Locale.Tr "repo.activity.period.halfweekly"}}</a>
+			<a class="{{if eq .Period "weekly"}}active {{end}}item" href="{{$.RepoLink}}/activity/weekly">{{ctx.Locale.Tr "repo.activity.period.weekly"}}</a>
+			<a class="{{if eq .Period "monthly"}}active {{end}}item" href="{{$.RepoLink}}/activity/monthly">{{ctx.Locale.Tr "repo.activity.period.monthly"}}</a>
+			<a class="{{if eq .Period "quarterly"}}active {{end}}item" href="{{$.RepoLink}}/activity/quarterly">{{ctx.Locale.Tr "repo.activity.period.quarterly"}}</a>
+			<a class="{{if eq .Period "semiyearly"}}active {{end}}item" href="{{$.RepoLink}}/activity/semiyearly">{{ctx.Locale.Tr "repo.activity.period.semiyearly"}}</a>
+			<a class="{{if eq .Period "yearly"}}active {{end}}item" href="{{$.RepoLink}}/activity/yearly">{{ctx.Locale.Tr "repo.activity.period.yearly"}}</a>
```

Comment:
Regarding link stability and future maintainability, it's probably a better idea to generate these links so that they 1. store the bounds statically and 2. use `/-/` as prefix.
What I mean is the following:
Instead of
`<url>/<owner>/<repo>/activity/semiyearly`,
what about creating the link as
`<url>/<owner>/<repo>/-/activity/between-<startdate in YYYY-mm-dd format>-and<enddate in the same format>`
?

As an example, assume we have the owner `user` in the repo `repo`, then the route would look as follows:
`/user/repo/-/activity/between-2023-01-02-and-2024-01-02`.

This has three immediate benefits:
1. Users can easily query a custom span
2. Users know exactly which dates are contained
3. We don't care what your start day of the week is and can omit all these computations and assumptions. Instead, your week simply starts on the first day of the interval (so, whatever day of the week `2023-01-02` was in this example)

Reply:
Well, this has nothing to do with this PR. The "Pulse" page is formerly known as "Activity" page. The links were there and I copy pasted from there. See: https://github.com/go-gitea/gitea/blob/3084c990b04aea16278f012da38152ada1290a0d/templates/repo/activity.tmpl#L14-L20

Reply:
Well… Yes and no.
Yes, you are right that it is previous technical debt.
But no, it influences your PR insofar as you suddenly need to calculate week data, for which you first need to calculate when the week started.
With the proposed approach, you can omit most of these computations.
In particular, https://github.com/go-gitea/gitea/pull/27882/files#diff-c185f70ea2fdd520a70549958b839987758c95a37aacf65d666c974a199a4ea3R184-R258 should shrink quite a bit in this case.